code/dynamic_programming/src/factorial/factorial.java

```diff
@@ -1,23 +1,16 @@
-/* Part of Cosmos by OpenGenus Foundation */
+// factorial of a number by using dynami programming // contributed by deepyadav442
 
-/**
- * Implements factorial using recursion
- */
-public class Factorial {
-
-    private static int factorial(int num) {
-        if (num == 0) {
-            return 1;
-        } else {
-            return (num * factorial(num - 1));
-        }
-    }
 
+public class factorial {
     public static void main(String[] args) {
-        int number = 5;
-        int result;
-        result = factorial(number);
-        System.out.printf("The factorial of %d is %d", number, result);
-    }
+        long number = 7;
```

Comment:
Is there any way to extract this behavior to a function expecting a parameter for the number to take the factorial of? Makes it easier to use.